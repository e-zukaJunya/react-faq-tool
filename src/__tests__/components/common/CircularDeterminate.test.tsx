import React from "react";
import { shallow, mount } from "enzyme";
import CircularDeterminate from "components/common/CircularDeterminate";
import renderer, { act } from "react-test-renderer";

describe("renders CircularDeterminate", () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });
  afterEach(() => {
    jest.useRealTimers();
  });

  it("first", async () => {
    const wrapper = renderer.create(<CircularDeterminate />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  // useEffectのsetInterval実行
  it("after 1000ms", async () => {
    const wrapper = renderer.create(<CircularDeterminate />);
    await act(async () => {
      // Progressが100になるルートを確かめるため3000
      jest.runTimersToTime(3000);
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  // useEffctのreturn実行
  it("unmount", async () => {
    const wrapper = renderer.create(<CircularDeterminate />);
    wrapper.unmount();
    // clearIntervalが何かNumber型の値を引数に呼ばれたことをチェック
    expect(clearInterval).toHaveBeenCalledWith(expect.any(Number));
  });
});
