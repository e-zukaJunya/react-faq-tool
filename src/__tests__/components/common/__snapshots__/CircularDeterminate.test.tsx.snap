// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CircularDeterminate after 1000ms 1`] = `
<div
  className=""
>
  <div
    aria-valuenow={49}
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "transform": "rotate(262.710deg)",
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={
          Object {
            "strokeDasharray": "126.920",
            "strokeDashoffset": "33.012px",
          }
        }
      />
    </svg>
  </div>
</div>
`;

exports[`renders CircularDeterminate first 1`] = `
<div
  className=""
>
  <div
    aria-valuenow={0}
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "transform": "rotate(0.000deg)",
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={
          Object {
            "strokeDasharray": "126.920",
            "strokeDashoffset": "126.920px",
          }
        }
      />
    </svg>
  </div>
</div>
`;
